solana/solana-normalization/src/mapping.ts

```diff
@@ -396,7 +397,14 @@ class InstructionParser {
             address[stackHeight - 1] = 0
         }
 
+        if (stackHeight === 1) {
+            this.innerPos = -1
+        } else {
+            this.innerPos++
+        }
+
         let i: Instruction = {
+            innerInstructionIndex: this.innerPos,
```

Comment:
I think, it is better to set it directly, rather than recalculating it here.

Reply:
do you mean outside of `push` method? it seems that inner instructions can be created in several methods so it's hard to track them 🤷‍♂️